wolfcrypt/src/ecc.c

```diff
@@ -2490,6 +2490,65 @@ int wc_ecc_get_curve_size_from_id(int curve_id)
     return ecc_sets[curve_idx].size;
 }
 
+/* Returns the curve size that corresponds to a given curve name,
+ * as listed in ecc_sets[] of ecc.c.
+ *
+ * name    curve name, from ecc_sets[].name in ecc.c
+ * return  curve size, from ecc_sets[] on success, negative on error
+ */
+int wc_ecc_get_curve_size_from_name(const char* curveName)
+{
+    int x;
+
+    if (curveName == NULL)
+        return BAD_FUNC_ARG;
+
+    /* find curve from name */
+    for (x = 0; ecc_sets[x].size != 0; x++) {
+        if (XSTRNCMP(ecc_sets[x].name, curveName,
+                    XSTRLEN(curveName)) == 0) {
```

Comment:
Hey Chris, just a couple thoughts:

1. Recommend doing XSTRLEN() above as local variable, so it doesn't have to be done multiple times.
2. Recommend adding a case insensitive strnicmp support to our library and using that here.
3. We probably need to support "PRIME256V1" as a match for "SECP256R1".
4. To reduce duplicate code I'd add a third public function to get the ecc_set index for a curveName. Then call it in these two new functions.